data-sharing/R/utils.R

```diff
@@ -105,22 +113,51 @@ fetch_station_data <- function(last_measurements) {
 #' @return Data frame containing measurement data for the specified station,
 #'   or NULL if query fails
 get_station_measurements <- function(date, station_id) {
-  
   time_param <- glue::glue("last_updated_lvl_time>={date}")
   station_id_param <- glue::glue('station_id="{station_id}"')
-  
-  logger::log_info("Querying '{dataset_id}' dataset for {station_id} since {date}")
-
-  tryCatch({
-    rerddap::tabledap(
-      dataset_id,
-      fields = columns,
-      fmt = 'parquet',
-      time_param,
-      station_id_param
+
+  logger::log_info(
+    "Querying '{dataset_id}' dataset for {station_id} since {date}"
+  )
+
+  tryCatch(
+    {
+      rerddap::tabledap(
+        dataset_id,
+        fields = columns,
+        fmt = 'parquet',
+        time_param,
+        station_id_param
+      )
+    },
+    error = function(e) {
+      logger::log_error(
+        "Failed to fetch data for station {station_id}: {e$message}"
+      )
+      return(NULL)
+    }
+  )
+}
+
+
+capture_sentry_exception <- function(e) {
+  logger::log_info("Creating sentry alert.")
+  if (is_main() && is_gha()) {
```

Comment:
this is nice, better then our current method of relying on an environment variable